src/openloco/localisation/stringmgr.cpp

```diff
@@ -283,43 +283,43 @@ namespace openloco::stringmgr
                 {
                     case control_codes::int32_grouped:
                     {
-                        int32_t value = args.pop32();
+                        int32_t value = args.popS32();
                         buffer = format_int32_grouped(value, buffer);
                         break;
                     }
 
                     case control_codes::int32_ungrouped:
                     {
-                        int32_t value = args.pop32();
+                        int32_t value = args.popS32();
                         buffer = format_int32_ungrouped(value, buffer);
                         break;
                     }
 
                     case control_codes::int16_decimals:
                     {
-                        int16_t value = args.pop16();
+                        int16_t value = args.popS16();
                         buffer = format_short_with_decimals(value, buffer);
                         break;
                     }
 
                     case control_codes::int32_decimals:
                     {
-                        int32_t value = args.pop32();
+                        int32_t value = args.popS32();
                         buffer = format_int_with_decimals(value, buffer);
                         break;
                     }
 
                     case control_codes::int16_grouped:
                     {
-                        int16_t value = args.pop16();
+                        int16_t value = args.popS16();
                         buffer = format_int32_grouped(value, buffer);
                         break;
                     }
 
                     case control_codes::uint16_ungrouped:
                     {
-                        uint16_t value = args.pop16();
-                        buffer = format_int32_ungrouped((int32_t)value, buffer);
+                        int32_t value = args.pop16();
```

Comment:
The control code says `uint16`. Should either be adjusted?

Reply:
It pops a `uint16`, and then stores it into a 32-bit variable. So no bits get lost

Reply:
Ah, right. Fair enough.